flamepool_test.go

```diff
@@ -165,6 +165,21 @@ func TestResultChannelReception(t *testing.T) {
 	}
 }
 
+func TestClose(t *testing.T) {
+	elements := []string{"rulo", "tomcat"}
+	pool := New(1, elements)
+
+	pool.Close()
+
+	select {
+	case <-pool.errorChan:
+	case <-pool.innerChan:
+	case <-pool.resultChan:
+	default:
+		t.Error("Channels should be closed")
```

Comment:
The t.Error shouldn't be in each case?

Reply:
If all channels are closed, the select statement is ignored causing the test to pass. 
If any channel is open, and since no message is sent to any channel it should execute the default statement raising t.Error().

This test would fail to verify if all channels are closed, if a message is sent to any open channel exactly at the moment this test is executed, which is _very_ unlikely.

Reply:
Makes sense!